lib/flatpak-installation.c

```diff
@@ -920,20 +975,104 @@ flatpak_installation_list_installed_refs_for_update (FlatpakInstallation *self,
         g_ptr_array_add (updates, g_object_ref (installed_ref));
     }
 
-  return g_steal_pointer (&updates);
-}
-
 #ifdef FLATPAK_ENABLE_P2P
-static void
-async_result_cb (GObject      *obj,
-                 GAsyncResult *result,
-                 gpointer      user_data)
-{
-  GAsyncResult **result_out = user_data;
-  *result_out = g_object_ref (result);
-}
+  context = g_main_context_new ();
+  g_main_context_push_thread_default (context);
+
+  collection_refv = (OstreeCollectionRef **) g_ptr_array_free (collection_refs, FALSE);
+  collection_refs = NULL;
+  ostree_repo_find_remotes_async (flatpak_dir_get_repo (dir),
+                                  (const OstreeCollectionRef * const *) collection_refv,
+                                  NULL,  /* no options */
+                                  NULL, /* default finders */
+                                  NULL,  /* no progress */
+                                  cancellable,
+                                  async_result_cb,
+                                  &result);
+
```

Comment:
Will this not also find the "regular" remote updates. I.e. are we not finding everything twice?
It seems like if we're doing this, then we could skip the flatpak_installation_list_remote_refs_sync() calls?

Reply:
For remotes with collection IDs set, we do skip the flatpak_installation_list_remote_refs_sync() calls. And ostree_repo_find_remotes_async() is only checking refs from remotes with collection IDs set.